gio/src/cancellable.rs

```diff
@@ -1,10 +1,153 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
 use crate::prelude::*;
+use crate::Cancellable;
+use futures_channel::oneshot;
+use futures_core::Future;
+use glib::{translate::*, IsA, Source};
 
-#[test]
-fn check_callback() {
-    let c = crate::Cancellable::new();
-    c.connect_cancelled(|_| {});
-    c.cancel(); // if it doesn't crash at this point, then we're good to go!
+// rustdoc-stripper-ignore-next
+/// The id of a cancelled handler that is returned by `CancellableExtManual::connect`. This type is
+/// analogous to `glib::SignalHandlerId`.
+#[derive(Debug, Eq, PartialEq)]
+#[repr(transparent)]
+pub struct CancelledHandlerId(libc::c_ulong);
```

Comment:
I'd make this a `NonZeroU64`. Basically, mirror how `SignalHandlerId` works

Reply:
I used to have it as that but it broke the tests, zero is actually a valid return value here: https://docs.gtk.org/gio/method.Cancellable.connect.html#return-value